### ISSUE TITLE
Ceres future / Maintainer needed ?

### ISSUE THREAD

**Issue by deniszh** [graphite-project/ceres#45]
Hi all,

We need to discuss Ceres future. As far as I understand, it still requires `megacarbon` branch and it seems it's not `intended to replace Whisper as the default storage format for Graphite` anymore.
So, I see a couple of options here:
1) find some active Ceres user, who can help with merging Ceres support to master and fix (or backport from an own branch) all fixes. (Maybe we can include Ceres as pluggable backend then?) 🚢 
2) Just admit current state, put `megacarbon` branch in same repo as ceres as an archive, and officially abandon that. 🔥 
3) Do nothing and continue to confuse users. 🙄 
Let's discuss here.
/cc @cbowman0 @iain-buclaw-sociomantic 
/cc @obfuscurity  @DanCech @iksaif 

If someone knows some Ceres' users - please show them this thread.

**Comment by iksaif** [comment]
My highly biased opinion would be to deprecate Ceres in favor of BigGraphite.

**Comment by iain-buclaw-sociomantic** [comment]
Just to repeat what I said in #31 here.

1. Carbon has full ceres support - graphite-project/carbon#484, graphite-project/carbon#508, and graphite-project/carbon#596.

2. Megacarbon has performance issues - it consumes about twice the CPU usage.  As far as I'm concerned, megacarbon is dead, long live megacarbon!   :fire: 

3. I think the confusion caused may be in part a fault of lack of PR (documentation, blog posts, etc).  Probably not helped by that the patches I submitted to master were very much under the radar.  

There's also been no release of carbon-master in over two years?  This probably is not helping exposure either.

**Comment by deniszh** [comment]
> My highly biased opinion would be to deprecate Ceres in favor of BigGraphite.

I think it's not mutually exclusive tasks. IMO "Let a thousand flowers bloom". We need to implement proper plugging storage and finder support and let it live - if we have active users of it. And if we can find some maintainer ofc.

**Comment by deniszh** [comment]
> Megacarbon has performance issues - it consumes about twice the CPU usage. As far as I'm concerned, megacarbon is dead, long live megacarbon!

If Ceres working with current master - IMO we should kill `megacarbon`, indeed. If I'm not missing some other value there.

> There's also been no release of carbon-master in over two years? This probably is not helping exposure either.

We'll fix it real soon, I hope. :)

**Comment by iksaif** [comment]
If we don't have any maintainer for Ceres, maybe we should mark it clearly as un-maintained in the README / description ? That would make things less confusing

**Comment by deniszh** [comment]
Yes, I changed my initial post. If we'll not find maintainer in near future I'm going to do exactly that.

**Comment by iain-buclaw-sociomantic** [comment]
> My highly biased opinion would be to deprecate Ceres in favor of BigGraphite.

I think my counter opinion would be that if carbon can be made more pluggable, that should be made to happen.

There for sure are benefits of using Ceres over Whisper despite the trade-off of having roll-up done as an offline operation (e.g: nightly cron).  But there's probably nothing more that can be changed in Ceres to be improved upon on the database level.  The space saving is nice, but I don't think it goes far enough in being a radical alternative.

The next place I'd like to be would be something along the lines of storing timestamps as delta of deltas.  Then offload the "normalizing" of data to fixed intervals to graphite.  However I'm sure there are already engines out there that do this, I just haven't discovered them yet. :-)

**Comment by iksaif** [comment]
See #46
